src/averageAges.js

```diff
@@ -55,7 +78,16 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let kids = people.filter(kid => kid.mother
+    && people.find(human => human.name === kid.mother));
+
+  kids = !onlyWithSon
+    ? kids
+    : kids.filter(kid => kid.sex === 'm');
+
+  return kids.reduce((acc, cur) =>
+    acc + cur.born - people.find(human =>
+      human.name === cur.mother).born, 0) / kids.length;
```

Comment:
```suggestion
  return kids.reduce((acc, cur) => (
    acc + cur.born - people.find(
      human => human.name === cur.mother
    ).born, 0)
  ) / kids.length;
```

Reply:
But better to save result of find to variable first

Reply:
![image](https://user-images.githubusercontent.com/79811705/191701502-f4c9734a-0eda-4c18-9a59-3169bf48d21c.png)
lint arguing when i do like you suggest.
"Unexpected use of comma operator".
i will do almost like you suggest but without extra hooks.